mindmeld/components/dialogue.py

```diff
@@ -988,6 +1012,18 @@ def __call__(self, request, responder):
             # Fill the form with the entities in the first query
             self._initial_fill(request)
 
+        # convert json response to FormEntity objects (deserialize)
+        for i, slot in enumerate(self._local_entity_form):
+            if not isinstance(slot, FormEntity):
```

Comment:
It looks like we assume slot is a dict here, why dont we just check for `if not isinstance(slot, dict):` instead?